### PR TITLE
Adds `--with-maintenance-log` option

### LINKED ISSUE #4688: Add `--with-maintenance-log` to "disposeOutdatedEntities.php"
Refs: #4484

### Issue
It well be nice to be able to enable event logging with the additional `--with-maintenance-log` option to the "disposeOutdatedEntities.php" script like it is done for e.g. "rebuildData.php".

Same as https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/1764 but for this script.

### PR DESCRIPTION
**Pull request by kghbln**
This PR is made in reference to: #4688

This PR addresses or contains:
- Adds `--with-maintenance-log` option

This PR includes:
- [ ] Tests (unit/integration)
- [x] CI build passed

Fixes #4688

### CODE REVIEW

**Comment by kghbln** [comment]
@mwjames A lot of guessing involved.

File: maintenance/disposeOutdatedEntities.php [hunk 1]
```diff
@@ -106,6 +112,11 @@ public function execute() {
 		$outdatedDisposer->setMessageReporter( $this->messageReporter );
 		$outdatedDisposer->run();
 
+		if ( $this->hasOption( 'with-maintenance-log' ) ) {
+			$maintenanceLogger = $maintenanceFactory->newMaintenanceLogger( 'DisposeOutdatedEntitiesLogger' );
+			$maintenanceLogger->log( $maintenanceHelper->transformRuntimeValuesForOutput() );
```
Comment (mwjames) [inline maintenance/disposeOutdatedEntities.php:117]: `transformRuntimeValuesForOutput` no longer exists which got replaced by something like:

```
			$runtimeValues = $maintenanceHelper->getRuntimeValues();

			$log = [
				'Memory used' => $runtimeValues['memory-used'],
				'Time used' => $runtimeValues['humanreadable-time']
			];

			$maintenanceLogger->logFromArray( $log );
```

Which allows to add other information to the $log array without having the `$maintenanceHelper` to pull-in script specific information.

Comment (kghbln) [reply maintenance/disposeOutdatedEntities.php:117]: Ah, I saw this but figured that it was specific to that respective script. Thanks for the note.

**Comment by kghbln** [comment]
[Documented](https://www.semantic-mediawiki.org/w/index.php?title=Help%3AMaintenance_script_disposeOutdatedEntities.php&type=revision&diff=75985&oldid=75489)

**Pull request merged**